mobile/src/main/java/com/google/samples/apps/iosched/util/initializers/AnalyticsHelperInitializer.kt

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.samples.apps.iosched.util.initializers
+
+import android.content.Context
+import androidx.startup.Initializer
+import com.google.samples.apps.iosched.di.InitializerEntryPoint
+import com.google.samples.apps.iosched.shared.analytics.AnalyticsHelper
+import javax.inject.Inject
+
+class AnalyticsHelperInitializer : Initializer<Unit> {
+
+    @Inject
+    lateinit var analyticsHelper: AnalyticsHelper
+
+    override fun create(context: Context) {
+        InitializerEntryPoint.resolve(context).inject(this)
```

Comment:
```suggestion
        analyticsHelperInitializer = InitializerEntryPoint.resolve(context). analyticsHelperInitializer()
```

As commented above, I'd rather not have entry points behave like Dagger components. Also, the `@Inject` from `analyticsHelper` can be removed and can be made `private`.

Thanks!

Reply:
Sorry, I didn't get this, why `EntryPoint` returns `initializer` instance instead of `dependency` which is `AnalyticsHelper` in this class?

Reply:
Yeah, sorry, meant `analyticsHelper = InitializerEntryPoint.resolve(context). analyticsHelper()` instead